client/state/posts/selectors.js

```diff
@@ -20,10 +32,40 @@ import { DEFAULT_POST_QUERY } from './constants';
  * @return {Object}          Post object
  */
 export function getPost( state, globalId ) {
-	return state.posts.items[ globalId ];
+	return getPostsByGlobalId( state )[ globalId ];
+}
+
+/**
+ * Returns an array of all posts.
+ *
+ * @param  {Object} state    Global state tree
+ * @return {Array}           Posts array
+ */
+export function getPosts( state ) {
+	return state.posts.items;
 }
 
 /**
+ * Returns a mapping of site ID, post ID pairing to global post ID.
+ *
+ * @param state
+ */
+export const getSitePosts = createSelector(
```

Comment:
In my mind, this function would return an array of posts for a particular site ID, passed as an argument. This does not seem straight-forward to achieve in `reselect` as they [(intentionally)](https://github.com/reactjs/reselect#q-how-do-i-create-a-selector-that-takes-an-argument) do not make it easy to create memoized selectors which accept arguments. Have you thought through how this might work? Perhaps a separate function which calls on the memoized site posts selector (`getSitePosts`, `getPostsBySiteId`)?

To the earlier discussion, I'd like to think that `reselect` selectors could live alongside our "pure functions" approach, but some of the opinions of the library clash with ours enough that I worry about the confusion it might cause to treat them as the same.